File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x == 0) {
            return 0;
        }

        int low = 1;
        int high = x;
        int ans;

        while (low <= high) {
            long long int mid = low + (high - low) / 2;

            long long int sqr = mid * mid;

            if (sqr == x) {
                return mid;
            } else if (sqr < x) {
                ans = mid;
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }

        return ans;
    }
};

File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if(x==0)
        {
            return 0;
        }
        int start=0;
        int end=x;
        int ans=0;
        while(start<=end)
        {
            int mid=(start+end)/2;
            long sqr=(long) mid*mid;
            if(sqr==x)
            {
                ans=mid;
                return ans;
            }
            else if(sqr<x)
            {
                start=mid+1;
                ans=mid;
            }
            else
            {
                end=mid-1;
            }
        }
        return ans;
    }
}